### DISCUSSION TITLE
Not possible to show a field that is in a group as a column

### DISCUSSION THREAD

**Discussion by DavidOliver** [payloadcms/payload#1617] [Feature Requests & Ideas]
# Bug Report

I've been separating fields into groups to visually break up the CRUD interface but have now found that that precludes child fields being individually shown as columns in the listing view.

For example, I have a group named 'details' containing fields named 'nameFirst' and 'nameLast':

![image](https://user-images.githubusercontent.com/192853/206244248-9be749f4-5d98-4c0b-80e6-2d357377fdb0.png)

![image](https://user-images.githubusercontent.com/192853/206244415-9e0a389a-ae3b-421e-8d60-29d8fea27a45.png)

![image](https://user-images.githubusercontent.com/192853/206243010-c87a2364-450e-4ce1-bf68-9bfa55c0a7ac.png)

## Steps to Reproduce

1. Create a group field that contains other fields
2. Show the Columns selector on the listing page

## Other Details

Payload `1.2.5`

**Comment by jacobsfletch** [comment]
This is currently not possible but would make a great feature. In the meantime, you could inject your own `Cell` component and display whatever you'd like. https://payloadcms.com/docs/admin/components#cell-component

**Comment by DavidOliver** [comment]
Thanks - that's good to know.

I think for now I'll need to remove the use of groups as it's not ([yet](https://github.com/payloadcms/payload/discussions/1498)) straightforwardly possible to prevent groups from showing the columns selector, and my client is bound to find that out and mention it. :)

It would be great to have some way to visually break up sets of fields/add spacing that didn't affect the fields' names/grouping behaviour.

**Comment by JarrodMFlesch** [comment]
@DavidOliver you could implement this yourself if you wanted to by rearranging your config fields!

Personally I would wire up re-usable `ui` react component and render that between `row` fields (rows do not affect data shape). The UI field could take a title and add top padding/margin with a top border to get a similar look and feel of the group component 😃 

Edit: but we are all in agreement that the searching/sorting/filtering could use some love, for sure 👍

**Comment by DavidOliver** [comment]
Thanks, both.

> This is currently not possible but would make a great feature. In the meantime, you could inject your own Cell component and display whatever you'd like. https://payloadcms.com/docs/admin/components#cell-component

I don't have any experience with React. I have the example cell component replacing the whole list view by using it with `admin.components.views.List`.

```ts
import { CollectionConfig } from 'payload/types';
import React from "react";

const baseClass = "custom-cell";

const CustomCell: React.FC<Props> = (props) => {
  const { field, colIndex, collection, cellData, rowData } = props;

  return <span className={baseClass}>{cellData}</span>;
};

const Systems: CollectionConfig = {
  slug: 'systems',
  admin: {
    components: {
      views: {
        List: CustomCell,
      },
    }
  },
```

And, of course, this results in an essentially blank view.

How can I instead inject it into the default list view (and only for the group field concerned?)?

If this kind of question is not appropriate here, I understand. I can simply remove the groups for now and come back to this later.

**Comment by DanRibbens** [comment]
You could replace your field `group` with `collapsible` - https://payloadcms.com/docs/fields/collapsible.

That is organizational only so your data shape will change slightly.

I'm converting this to a discussion so we can track it as a new feature.

**Comment by JarrodMFlesch** [comment]
@DavidOliver, I think what you are looking to do is slightly different than what you have above. Something like this is what I was thinking if you are just looking to get the Cell in the table to look nice:
```typescript
{
  type: 'group',
  name: 'groupField',
  admin: {
    components: {
      Cell: CustomCell
    },
  },
  fields: [ ...yourFieldsHere ],
}
  ```

**Comment by DavidOliver** [comment]
@JarrodMFlesch, thanks very much - that gets me further. It looks like the relationship fields' values in `cellData` within the group are the IDs rather than the string names of the related entries. It's probably more practical for me to simplify and do without the groups.

@DanRibbens, thanks - I think collapsibles will do the job. It's nice that they can initially appear not collapsed.

**Comment by cbratschi** [comment]
I checked the redirects plugin and it uses a group reference for 'to.type':

https://github.com/payloadcms/plugin-redirects/blob/main/src/index.ts#L16

My implementation uses now two groups: from and to. All data is stored there and therefore cannot be shown in columns. Would be great if this would be supported. I guess I have to remove the groups again to get the columns working.

**Comment by andershermansen** [comment]
This is implemented in #7355
